pyvrp/GeneticAlgorithm.py

```diff
@@ -146,11 +152,14 @@ def _educate(self, individual: Individual):
             and self._rng.rand() < self._params.repair_probability
         ):
             with self._pm.get_penalty_booster() as booster:  # noqa
-                self._ls.search(individual)
+
+                # TODO shouldn't we resample the 'intensify' variable here?
+                # this is what HGS used to do
+                self._ls.run(individual, intensify)
 
                 if individual.is_feasible():
                     if (
-                        self._params.should_intensify
+                        self._params.intensify_on_best
```

Comment:
We do the intensify before before saving the new best, whereas the individual in theory may become infeasible by intensification so shouldn't we check and store the new best both before and after doing this intensifcation?